worker/worker-lib.ts

```diff
@@ -0,0 +1,59 @@
+import { v4 as uuid } from 'uuid';
+
+// @ts-ignore
+import UMAPWorker from './umap.worker';
+
+const requests = {};
+
+function onError(id, error) {
+  console.error(error);
+  try {
+    requests[id].reject(error.message);
+  } finally {
+    delete requests[id];
+  }
+}
+
+function onMessage({ data: { error, id, method, results } }) {
+  if (error) {
+    onError(id, new Error(error));
+  }
+  switch (method) {
+    case 'fit': {
+      const { embedding } = results;
+      if (requests[id]) {
+        requests[id].resolve(embedding);
+      } else {
+        requests[id].reject(
+          `Could not find existing UMAPWorker request with id ${id}`
+        );
+      }
+      delete requests[id];
+      break;
+    }
+    default: {
+      throw new Error(
+        `Unknown method performed by UMAPWorker: 'umap.${method}'`
+      );
+    }
+  }
+}
+
+const worker = new UMAPWorker();
```

Comment:
Creating a worker instance should be in the constructor of the `UMAPWorkerWrapper` but I was failing to type it correctly and moved it out to move forward with the proof-of-concept 🙃

Reply:
👍